src/Parser.zig

```diff
@@ -797,12 +797,10 @@ fn decl(p: *Parser) Error!bool {
         }
         switch (p.tok_ids[first_tok]) {
             .asterisk, .l_paren, .identifier, .extended_identifier => {},
-            else => if (attr_buf_top == p.attr_buf.items.len) {
-                return false;
-            } else {
+            else => if (attr_tokens_seen) {
```

Comment:
Do you think this check will be needed in the other places? Otherwise this check can just be `p.tok_i != first_tok`.

Reply:
I wasn't sure if `declSpec` could consume tokens without being successful, but it looks like it can't. Updated it to just do that comparison.